sdk/core/Azure.Core.Experimental/src/MutableJsonElement.ObjectEnumerator.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text.Json;
+
+namespace Azure.Core.Json
+{
+    /// <summary>
+    /// A mutable representation of a JSON element.
+    /// </summary>
+    public readonly partial struct MutableJsonElement
+    {
+        /// <summary>
+        ///   An enumerable and enumerator for the properties of a JSON object.
+        /// </summary>
+        [DebuggerDisplay("{Current,nq}")]
+        public struct ObjectEnumerator : IEnumerable<(string Name, MutableJsonElement Value)>, IEnumerator<(string Name, MutableJsonElement Value)>
+        {
+            private readonly MutableJsonElement _target;
+            private JsonElement.ObjectEnumerator _enumerator;
+
+            internal ObjectEnumerator(MutableJsonElement target)
+            {
+                Debug.Assert(target.ValueKind == JsonValueKind.Object);
```

Comment:
Is this intentionally only validating in Debug mode rather than `EnsureObject()`?

Reply:
Oh good catch, thank you.